share/secretsmanager.py

```diff
@@ -0,0 +1,167 @@
+            raise SyntaxError("Invalid arn format: {}".format(secret_arn))
+
+        if secret_arn not in config_secret_entry_values:
+            config_secret_entry_values[secret_arn] = ""
+
+        region = splitted_secret_arn[3]
+        secrets_manager_name = splitted_secret_arn[6]
+
+        if region == "":
+            raise ValueError("Must be provided region in arn: {}".format(secret_arn))
+
+        if secrets_manager_name == "":
+            raise ValueError("Must be provided secrets manager name in arn: {}".format(secret_arn))
+
+        if secrets_manager_name not in secret_consistency_len_check:
+            secret_consistency_len_check[secrets_manager_name] = len(splitted_secret_arn)
+        else:
+            if secret_consistency_len_check[secrets_manager_name] != len(splitted_secret_arn):
+                raise ValueError(
+                    "You cannot have both plain text and json key for the same secret: {}".format(secret_arn)
+                )
+
+        if region not in secret_key_values_cache:
+            secret_key_values_cache[region] = {}
+
+        if secrets_manager_name not in secret_key_values_cache[region]:
+            secret_key_values_cache[region][secrets_manager_name] = {}
+
+    for region in secret_key_values_cache:
+        for secrets_manager_name in secret_key_values_cache[region]:
+            str_secrets = get_secret_values(secrets_manager_name)
+            parsed_secrets = parse_secrets_str(str_secrets)
+
+            secret_key_values_cache[region][secrets_manager_name] = parsed_secrets
+
+    for config_secret_entry in config_secret_entry_values:
+        splitted_secret_arn = config_secret_entry.split(":")
+
+        region = splitted_secret_arn[3]
+        secrets_manager_name = splitted_secret_arn[6]
+
+        if len(splitted_secret_arn) == 8:
+            wanted_key = splitted_secret_arn[-1]
+            if wanted_key == "":
+                raise ValueError("Key must not be empty: {}".format(config_secret_entry))
+
+        if (
+            isinstance(secret_key_values_cache[region][secrets_manager_name], dict)
+            and wanted_key in secret_key_values_cache[region][secrets_manager_name]
+        ):
+            fetched_secret_entry_value = secret_key_values_cache[region][secrets_manager_name][wanted_key]
+            if fetched_secret_entry_value == "":
+                raise ValueError("Value for key: {} must not be empty".format(config_secret_entry))
+            config_secret_entry_values[config_secret_entry] = fetched_secret_entry_value
+        elif (
+            isinstance(secret_key_values_cache[region][secrets_manager_name], dict)
+            and wanted_key not in secret_key_values_cache
+        ):
+            raise KeyError("Key: {} not found in: {}".format(wanted_key, secrets_manager_name))
+        else:
+            config_secret_entry_values[config_secret_entry] = secret_key_values_cache[region][secrets_manager_name]
```

Comment:
here `secret_key_values_cache[region][secrets_manager_name]` could be empty like for https://github.com/elastic/elastic-serverless-forwarder/pull/42/files#diff-db1d9ebcb636e71a1e8c74b68c0eb2aff1e7ca7e304450f805cbfcd7011c2ed2R99

we should raise a `ValueError` as well